backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -438,16 +438,20 @@ def handle_revoke_failure(self, error: Exception) -> bool:
         -------
         True if alarm published successfully
         """
-        logger.error(
-            f'Failed to revoke S3 permissions to bucket {self.bucket_name} '
-            f'from source account {self.source_environment.AwsAccountId}//{self.source_environment.region} '
-            f'with target account {self.target_environment.AwsAccountId}/{self.target_environment.region} '
-            f'due to: {error}'
-        )
-        DatasetAlarmService().trigger_revoke_folder_sharing_failure_alarm(
-            self.target_bucket, self.share, self.target_environment
-        )
-        return True
+        try:
+            logger.error(
+                f'Failed to revoke S3 permissions to bucket {self.bucket_name} '
+                f'from source account {self.source_environment.AwsAccountId}//{self.source_environment.region} '
+                f'with target account {self.target_environment.AwsAccountId}/{self.target_environment.region} '
+                f'due to: {error}'
+            )
+            DatasetAlarmService().trigger_revoke_s3_bucket_sharing_failure_alarm(
+                self.target_bucket, self.share, self.target_environment
+            )
+            return True
+        except Exception as e:
```

Comment:
Change to:

logger.error("Could not process dataset alarms: ", exc_info=True)
return False